### ISSUE TITLE
[Bug]: Redundant Calculation of Reduced Embeddings

### ISSUE THREAD

**Issue by Madhav-MKNC** [infiniflow/ragflow#1781]
### Is there an existing issue for the same bug?

- [ ] I have checked the existing issues.

### Branch name

main

### Commit ID

4ba1ba9

### Actual behavior

The UMAP dimensionality reduction is recalculated in each iteration of the loop, which could be optimized. An issue can be raised to cache results of the UMAP reduction if the embeddings do not change between iterations.

```
reduced_embeddings = umap.UMAP(
    n_neighbors=max(2, n_neighbors), n_components=min(12, len(embeddings)-2), metric="cosine"
).fit_transform(embeddings)
```

### Steps to reproduce

```Markdown
Optimize UMAP dimensionality reduction by caching results or reusing computations when possible to improve performance.
```

**Comment by hangters** [comment]
hi @Madhav-MKNC  thanks for mention thart . but it looks like there is no recalculated in each iteration , the embeddings is different in each iteration since the variable start and end would change in each iteration.
`embeddings = [embd for _, embd in chunks[start: end]]
......
reduced_embeddings = umap.UMAP(
    n_neighbors=max(2, n_neighbors), n_components=min(12, len(embeddings)-2), metric="cosine"
).fit_transform(embeddings)
`
